api/i18n.js

```diff
@@ -0,0 +1,150 @@
+import pino from "pino";
+import * as cliApi from "@lingui/cli/api";
+import { relative, join } from "node:path";
+import { writeFileSync } from "node:fs";
+import Table from "cli-table";
+import PathResolver from "../lib/path.js";
+import chalk from "chalk";
+
+function getStats(catalog) {
+  return [Object.keys(catalog).length, Object.keys(catalog).filter((key) => !catalog[key].translation).length];
+}
+
+function printExtractionStats(config, catalogs) {
+  const table = new Table({
+    head: ["Language", "Total count", "Missing"],
+    colAligns: ["left", "middle", "middle"],
+    style: {
+      head: ["green"],
+      border: [],
+      compact: true,
+    },
+  });
+
+  Object.keys(catalogs).forEach((locale) => {
+    const catalog = catalogs[locale];
+    // catalog is null if no catalog exists on disk and the locale
+    // was not extracted due to a `--locale` filter
+    const [all, translated] = catalog ? getStats(catalog) : ["-", "-"];
+
+    if (config.sourceLocale === locale) {
+      table.push({ [`${locale} (source)`]: [all, "-"] });
+    } else {
+      table.push({ [locale]: [all, translated] });
+    }
+  });
+
+  return table;
+}
+
+/**
+ * Handle i18n extraction and compilation for the podlet
+ * @param {string} command - The command to run.
+ * @param {import("convict").Config} config - The podlet configuration.
+ * @param {string} cwd - The current working directory.
+ * @returns {Promise<void>}
+ */
+export async function i18n({ command, config, cwd = process.cwd }) {
+  const logger = pino({
+    transport: {
+      target: "../lib/pino-dev-transport.js",
+    },
+    // @ts-ignore
+    level: config.get("app.logLevel").toLowerCase(),
+  });
+
+  const resolver = new PathResolver({ cwd });
+  const resolvedContentFile = await resolver.resolve("./content");
+
+  const compileNamespace = resolvedContentFile.typescript ? "ts" : "es";
+
+  /**
+   * @type {import("@lingui/conf").LinguiConfig} lingui configuration object
+   */
+  const linguiConfig = {
+    locales: config.get("app.locales") || ["en-US", "nb"],
+    sourceLocale: "en",
+    rootDir: cwd,
+    format: "po",
+    catalogs: [
+      {
+        include: [join(cwd, "{content,fallback,document}.{ts,js}")],
```

Comment:
are files listed here the only files included? I see there is an exclude as well so maybe not? In any case, we need to include files inside folders named src or client off the top of my head at least, maybe others

Reply:
> are files listed here the only files included? I see there is an exclude as well so maybe not? In any case, we need to include files inside folders named src or client off the top of my head at least, maybe others

@digitalsadhu These are the only ones included now indeed. I'll make sure to include src and client folders.